Assets/Tests/InputSystem/CoreTests_State.cs

```diff
@@ -1014,6 +1014,25 @@ public unsafe void State_CanGetMetrics()
         Assert.That(metrics.averageProcessingTimePerEvent, Is.GreaterThan(0.0000001));
     }
 
+    [Test]
+    [Category("State")]
+    public void State_PressingAndReleasingKeyInSameFrame_DoesNotShowStateChange()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        var firstState = new KeyboardState(Key.A);
+        var secondState = new KeyboardState();
+
+        InputSystem.QueueStateEvent(keyboard, firstState);
+        InputSystem.QueueStateEvent(keyboard, secondState);
+
+        InputSystem.Update();
+
+        Assert.That(keyboard.aKey.isPressed, Is.False);
+        Assert.That(keyboard.aKey.wasPressedThisFrame, Is.False);
+        Assert.That(keyboard.aKey.wasReleasedThisFrame, Is.False);
+    }
```

Comment:
Wondering, does it make sense to have a separate test for this one given we have `State_PressingAndReleasingButtonInSameFrame_DoesNotShowStateChange`? The underlying cause is the same and the behavior applies to all buttons on any device.

Reply:
If you're certain that will always be the case, then I guess yes the test is redundant.

Reply:
Yup, certain it will be. Will/would be fixed for all ButtonControls on all devices if we find a good solution for the problem with the current front/back buffer mechanism.